installer/dev/MachineTypeAttributes.h

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+
+#include "pch.h"
+
+namespace MachineTypeAttributes
+{
+    using GetMachineTypeAttributesPtr = HRESULT(WINAPI*)(USHORT Machine, MACHINE_ATTRIBUTES* MachineTypeAttributes);
+
+    // On systems running Windows 11, this inline function returns if given architecture is supported on the system in user mode.
+    // On systems running down level Windows versions (ex: Windows 10), this inline function will return false, always.
+    inline bool IsWindows11_IsArchitectureSupportedInUserMode(USHORT architecture)
```

Comment:
NIT: Logic would be easier to read if it was written as fail/return/end-on-failure e.g.

```
dll = LoadLibrary
if (fail) THROW_BLAH(...)

p = GetProcAddress(...)
if (!p) THROW_LAST_ERROR()

THROW_IF_FAILED(getMachineTypeAttributes(architecture, &machineAttributes));
return WI_IsFlagSet(machineAttributes, MACHINE_ATTRIBUTES::UserEnabled);
```

Reply:
That's right, I agree with Throwing Load Library failure. But I shouldn't be throwing if the API isn't available because we use that currently to determine then that the system is Windows 10 and shouldn't install x64 packages on arm64. I will instead log the last error and add the error to existing telemetry in a separate field. Thanks for the cue.

By the way, IsExportPresent API in IsWindowsVersion.h follows the same pattern and doesn't throw any errors in loading library or not finding the API.

Reply:
_But I shouldn't be throwing if the API isn't available_
Right, but it should on errors. GetProcAddress returns null in both cases. We should distinguish Success-vs-Fail-vs-Error e.g.

```
if (!p)
{
     const auto lastError{ GetLastError() };
     THROW_HR_IF(HRESULT_FROM_WIN32(lastError), lastError!=i_forget_the_error_code_for_not_found)
}
```

>By the way, IsExportPresent API in IsWindowsVersion.h follows the same pattern and doesn't throw any errors in loading library or not finding the API.

Cool. That would be a bug :-)

An oversight when that was created. Feel free to improve it with the same sort of error logic you add in here